src/finch/algebra/operator.py

```diff
@@ -0,0 +1,17 @@
+import operator
+
+
+def and_test(a, b) -> bool:
+    return operator.truth(a) and operator.truth(b)
```

Comment:
The point of `and_test` was supposed to be the [prefix](https://www.cs.colostate.edu/~cs165/.Fall19/recitations/L15/doc/traversal-order.html) version of `and`, which is [infix](https://www.cs.colostate.edu/~cs165/.Fall19/recitations/L15/doc/traversal-order.html), similar to [Lib/operator.py](https://docs.python.org/3/library/operator.html). Please read all links in this comment completely, let me know if you have questions.

`operator` defines prefix versions of many infix python calls. `operator.and_` is bitwise, (c's `&`).

You should implement `all` using `reduce(operator.and_, map(operator.truth, x))`. Implement `and_test` to be equivalent to `and` (drop `truth` here), and do not use `and_test` in `all`.